pkg/kademlia/kademlia_test.go

```diff
@@ -64,7 +64,10 @@ func TestNewKademlia(t *testing.T) {
 	for i, v := range cases {
 		dir := filepath.Join(rootdir, strconv.Itoa(i))
 
-		ca, err := provider.NewCA(context.Background(), 12, 4)
+		ca, err := provider.NewCA(context.Background(), provider.NewCAOptions{
```

Comment:
should we just have a NewTestCa function that sets the difficulty and concurrency ? that way if it needs to change in the future we only need to make the change in one place.

Reply:
That sounds nice - it would need to be something we use consistently across packages to be worth it because there are many of these.

Reply:
I made a function like that for something before, probably just need to make it public and move it into a TestUtils package or something